src/_guides/language/coming-from/swift-to-dart.md

```diff
@@ -0,0 +1,2982 @@
+class Rectangle {
+ var left, top, width, height: Double;
+
+ init(left: Double, top: Double, width: Double, height: Double) {
+   self.left = left
+   self.top = top
+   self.width = width
+   self.height = height
+ }
+
+ // Define two calculated properties: right and bottom.
+ var right: Double {
+   get {
+     return left + width
+   }
+   set { width = newValue - left }
+ }
+
+ var bottom: Double {
+   get {
+     return top + height
+   }
+   set { height = newValue - top }
+ }
+}
+```
+
+### Abstract classes
+
+Dart has the concept of _abstract_ classes,
+something that isn't supported by Swift.
+Abstract classes can't be instantiated directly and
+can only be subclassed. This makes abstract classes
+useful to define interfaces (comparable to a protocol in Swift).
+ 
+Abstract classes often contain _abstract_ methods,
+which are method declarations that don't have an
+implementation. Non-abstract subclasses are forced
+to override these methods and provide an appropriate
+implementation. An abstract class can also contain methods
+with a default implementation.
+Subclasses inherit this implementation
+if they don't override these methods when
+extending the abstract class.
+
+To define an abstract class, use the `abstract` modifier.
+The following example declares an abstract class
+that has an abstract method and a method
+containing a default implementation:
+
+{:.include-lang}
+```dart
+// This class is declared abstract and thus can't be instantiated.
+abstract class AbstractContainer {
+  void updateChildren(); // Abstract method.
+
+  // Method with default implementation.
+  String toString() => "AbstractContainer";
+}
+```
+
+### Implicit interfaces
+
+In the Dart language, every class implicitly
+defines an interface containing all the
+instance members of the class and of any
+interface it implements. If you want to create
+a class `A` that supports class `B`'s API
+without inheriting `B`'s implementation,
+class `A` should implement the `B` interface. 
+
+Unlike Dart, Swift classes don't implicitly
+define an interface. The interface needs to be explicitly
+defined as a protocol and implemented by the developer.
+
+A class can implement one or more interfaces
+and then provide the APIs required by the interfaces.
+Both Dart and Swift have a different ways of
+implementing interfaces. For example:
+
+{:.include-lang}
+```dart
+abstract class Animal {
+  int getLegs();
+  void makeNoise();
+}
+
+class Dog implements Animal {
+  @override
+  int getLegs() => 4;
+
+  @override
+  void makeNoise() => print('Woof woof');
+}
+```
+
+{:.include-lang}
+```swift
+protocol Animal {
+   func getLegs() -> Int;
+   func makeNoise()
+}
+
+class Dog: Animal
+{
+  func getLegs() -> Int {
+    return 4;
+  }
+
+  func makeNoise() {
+    print("Woof woof"); 
+  }
+}
+```
+
+### Extending a class
+
+Class inheritance in Dart is very similar to Swift.
+In Dart, you can use `extends` to create a subclass,
+and `super` to refer to the superclass:
```

Comment:
Note for others (or myself): Would be nice to include an example of using super in the following examples